### PR TITLE
subscriber: add span status fields

### LINKED ISSUE #1346: [tracing-opentelemetry] Option to set status of span similar kind/name
### Crates

tracing-opentelemetry

### Motivation

Right now all correct spans/traces are left without status codes and status messages. I want to be able to set custom span status to follow opentelemetry semantic conventions.

### Proposal

Add option to pass status code and message similarly to span kind and name:

https://github.com/tokio-rs/tracing/blob/master/tracing-opentelemetry/src/subscriber.rs#L206

### Alternatives

Mark all traces without errors as OK:

https://github.com/tokio-rs/tracing/blob/master/tracing-opentelemetry/src/subscriber.rs#L529

@jtescher please let me know what do you think. It looks like you are main contributor of this crate

### PR DESCRIPTION
**Pull request by aym-v**
## Motivation

Allow users to set custom span status codes and messages to follow opentelemetry semantic conventions.

## Solution

Support the status code and status message fields in `OpenTelemetrySubscriber`

Proposal to close #1346

### CODE REVIEW

**Review by jtescher — APPROVED** [review]
Looks good thanks @aym-v

File: tracing-opentelemetry/src/subscriber.rs [hunk 1]
```diff
@@ -85,18 +87,22 @@ impl WithContext {
...
+        _ => None,
+    }
+}
+
+fn str_to_status_code(s: &str) -> Option<otel::StatusCode> {
+    match s.to_ascii_lowercase().as_str() {
+        "unset" => Some(otel::StatusCode::Unset),
+        "ok" => Some(otel::StatusCode::Ok),
+        "error" => Some(otel::StatusCode::Error),
+        _ => None,
```
Comment (hawkw) [inline tracing-opentelemetry/src/subscriber.rs:105]: similarly, i think we should prefer `eq_ignore_ascii_case` for performance reasons:
```suggestion
    match s {
        s if s.eq_ignore_ascii_case("unset") => Some(otel::StatusCode::Unset),
        s if s.eq_ignore_ascii_case("ok") => Some(otel::StatusCode::Ok),
        s if s.eq_ignore_ascii_case("error") => Some(otel::StatusCode::Error),
        _ => None,
```

**Review by hawkw — COMMENTED** [review]
i noticed some potential performance regressions that this change could cause. we should probably avoid using `str::to_ascii_lowercase` when simply testing for equality.

besides that, this looks good to me! thank you!

**Comment by aym-v** [comment]
@hawkw the code is updated, thanks! 
The wasm tests did not pass but I think it has nothing to do with the commit.

**Comment by hawkw** [comment]
> @hawkw the code is updated, thanks!
> The wasm tests did not pass in the CI but I think it has nothing to do with the commit.

Hmm, yeah, that looks like some kind of issue with a dependency, I'll see if I can figure that out.

**Comment by hawkw** [comment]
Looks like upstream change japaric/cast.rs#27 should fix the CI failure.

**Pull request merged**